src/main/java/athena/TaskList.java

```diff
@@ -96,23 +115,32 @@ public String getDescription(int index) {
      * @param taskNumber Position of task in task list
      * @return Task deleted
      */
-    public Task deleteTask(int taskNumber) {
-        Task taskToDelete = tasks.get(taskNumber);
-        tasks.remove(taskNumber);
+    public Task deleteTask(int taskNumber) throws IndexOutOfBoundsException {
+        Task taskToDelete=null;
+        int counter = -1;
+        int index=-1;
+        for (Task t : tasks) {
+            counter += 1;
```

Comment:
Just a nitpick: `counter++` can ignore if you don't care lol

Reply:
I gladly accept this nitpick